### ISSUE TITLE
404 when trying to fetch original source

### ISSUE THREAD

**Issue by helgatheviking** [kathyisawesome/woo3pd-helpscout#14]
>ERROR Client error: `GET https://api.helpscout.net/v2/conversations/1248406680/threads/3575090471/original-source` resulted in a `404 Not Found` response

**Comment by helgatheviking** [comment]
[ref](https://github.com/helpscout/helpscout-api-php/issues/241#issuecomment-672904794)

Need to check the webhook's payload for 
```
  "source" : {
    "type" : "email",
    "via" : "customer"
  },
```

Or use a `try/catch` on the `getSource()` call:
```
try {
    $source = $client->threads()->getSource($convoId, $threadId);
} catch (\GuzzleHttp\Exception\ClientException $e) {
    if ($e->getResponse()->getStatusCode() === 404) {
        // thread's source not available
    }
}
```

Like #15 we want to gracefully continue with processing. Any uncaught error kills the processing and you get the unformatted ticket in HS.

**Comment by helgatheviking** [comment]
Closed by 051b9e1
